src/components/ContributorTasks/ContributorTasks.test.tsx

```diff
@@ -0,0 +1,124 @@
+        title: 'Front End Developer',
+      },
+    ],
+  };
+
+  it('renders without crashing', () => {
+    render(<ContributorTasks {...props} />);
+
+    expect(screen.getByTestId('ContributorTasks')).toBeTruthy();
+  });
+
+  it('renders with default className', () => {
+    render(<ContributorTasks {...props} />);
+    const taskComp = screen.getByTestId('ContributorTasks');
+
+    expect(taskComp).toHaveClass('ContributorTasks');
+  });
+
+  it('renders with passed in className', () => {
+    render(<ContributorTasks {...props} />);
+    const taskComp = screen.getByTestId('ContributorTasks');
+
+    expect(taskComp).toHaveClass('test');
+  });
+
+  it('renders all tasks', () => {
+    render(<ContributorTasks {...props} />);
+    const tasks = screen.getAllByTestId('ContributorTasks--row');
+
+    expect(tasks.length).toBe(props.tasks.length);
+  });
+
+  it('renders rows with title', () => {
+    render(<ContributorTasks {...props} />);
+    const tasks = screen.getAllByTestId('ContributorTasks__task-title');
+
+    const validTitle = tasks.every((task, i) => task.textContent === props.tasks[i].title);
+
+    expect(validTitle).toBeTruthy();
+  });
+
+  it('renders rows with valid github issue link', () => {
+    render(<ContributorTasks {...props} />);
+    const tasks = screen.getAllByTestId('A');
+
+    const validClass = tasks.every((task) => {
+      return task.className.includes('ContributorTasks__issue-link');
+    });
+
+    const validHref = tasks.every((task, i) => {
+      const propTask = props.tasks[i];
+      const href = `https://github.com/thenewboston-developers/${propTask.repository}/issues/${propTask.issue_id}`;
+
+      return task.getAttribute('href') === href;
+    });
+
+    expect(validClass).toBeTruthy();
+    expect(validHref).toBeTruthy();
+  });
+
+  it('renders rows with valid github repository repository', () => {
+    render(<ContributorTasks {...props} />);
+    const tasks = screen.getAllByTestId('ContributorTasks__repository');
+
+    const validRepo = tasks.every((task, i) => {
+      return task.textContent === props.tasks[i].repository;
+    });
+
+    expect(validRepo).toBeTruthy();
+  });
+
+  it('renders rows with date compeleted', () => {
+    render(<ContributorTasks {...props} />);
+    const tasks = screen.getAllByTestId('ContributorTasks__date-completed');
+
+    const validDate = tasks.every((task, i) => {
+      const date = format(props.tasks[i].completed_date, 'L/d/yy');
+      return task.textContent === date;
+    });
+
+    expect(validDate).toBeTruthy();
+  });
+
+  it('renders rows with valid amount', () => {
+    render(<ContributorTasks {...props} />);
+    const tasks = screen.getAllByTestId('ContributorTasks__amount');
+
+    const validAmount = tasks.every((task, i) => {
+      const amount = props.tasks[i].amount_paid;
+      const formattedAmount = parseInt(amount, 10).toLocaleString();
+      return task.textContent === `+ ${formattedAmount}`;
+    });
```

Comment:
hmm can we actually test this explicitly writing our what the string should be (instead of using logic to create the string)?

The reason is this... one of the benefits of unit tests is that it is almost self-documenting our code. But if we add too much JS logic within the test, we lose the self-documentation aspects of it.

Another reason is that it shows us exactly what the results should look like. A future developer might come in and make edits to this component, sees that this test is failing, and then just copies and pastes the logic to the test. This might defeat the purpose of the test later down the road